sites/svelte-5-preview/src/routes/docs/content/01-api/05-imports.md

```diff
@@ -93,6 +93,37 @@ To prevent something from being treated as an `$effect`/`$derived` dependency, u
 </script>
 ```
 
+### `createRawSnippet`
+
+An advanced API designed for people building frameworks that integrate with Svelte, `createRawSnippet` allows you to create [snippets](/docs/snippets) programmatically for use with `{@render ...}` tags:
+
+```js
+import { createRawSnippet } from 'svelte';
+
+const greet = createRawSnippet((name) => {
+	return {
+		render: () => `
+			<h1>Hello ${name()}!</h1>
+		`,
+		setup: (h1) => {
+			$effect(() => {
+				h1.textContent = `Hello ${name()}!`;
+			});
+		}
+	};
+});
+```
+
+The `render` function is called during server-side rendering, or during `mount` (but not during `hydrate`, because it already ran on the server), and must return HTML representing a single element.
```

Comment:
I'm super late to the party here and I haven't been following the length discussions about this API, but - is there an insurmountable technical concern that results in us requiring that this is has single top-level element? Regular snippets don't work that way, right?

Reply:
Oh, is it so that we can pass the single top-level element to `setup`? I had missed that that was what the `h1` variable was the first time around. I would maybe suggest a different name so we don't suggest that it's automatically finding certain nodes. Maybe just call it `node`?

Reply:
If there were multiple top-level nodes, then when hydrating we would need to know how many nodes to hydrate, otherwise things will get out of sync. That would mean either we have to generate the HTML during hydration, in order to see how many top-level nodes there are before immediately discarding them (wasteful) or we need to infer it from the arity of `setup` (brittle) which would become required even if it didn't do anything. Just a lot simpler this way.

Good edit though, will change

Reply:
Makes sense :+1: